layouts/shortcodes/youtube-embed.html

```diff
@@ -1,7 +1,56 @@
-<iframe 
-  class="aspect-video w-full"
-  src="https://www.youtube.com/embed/{{ .Get 0 }}?rel=0" 
-  title="YouTube video player" 
-  frameborder="0" 
-  allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" 
-  allowfullscreen></iframe>
\ No newline at end of file
+<div id="youtube-player-{{ .Get 0 }}" data-video-id="{{ .Get 0}}" class="youtube-video aspect-video w-full"></div>
+{{ if $.Page.Scratch.Get "youtube-embed" | default "incomplete" | ne "setup" }} 
+{{ $.Page.Scratch.Set "youtube-embed" "setup" }}
```

Comment:
there's a global `page` function that gives you access to the page context, so no need to access the global scope with `$`

```suggestion
{{ if page.Scratch.Get "youtube-embed" | default "incomplete" | ne "setup" }} 
{{ page.Scratch.Set "youtube-embed" "setup" }}
```

Reply:
Ah! Good to know. Happy to accept the suggestion!